### ISSUE TITLE
XeroExceptionHandler

### ISSUE THREAD

**Issue by dldobbin** [XeroAPI/Xero-Java#148]
XeroExceptionHandler::handleBadRequest(String) swallows any exception that does not contain "ApiException".

For example, AccountingApi::getAccounts has a catch for IOExceptions. If the IOException does not contain "ApiException", the exception handler returns null, resulting in a NullPointerException and the loss of any useful information the IOException might have contained.

**Comment by SidneyAllen** [comment]
Hi @dldobbin 

Thanks for calling this out.  Can you share some code where I can force and IOException and test the behavior you describe?  If you have fix and want to submit a PR along with the test code that would be great too.

**Comment by dldobbin** [comment]
```
public static void main(String[] args) {
  XeroExceptionHandler xeroExceptionHandler = new XeroExceptionHandler();
  try {
    try {
      throw new IOException("Useful Information that will never be seen");
    } catch (IOException e) {
      throw xeroExceptionHandler.handleBadRequest(e.getMessage());
    }
  } catch (XeroApiException xae) {
    //We never get here
  }
}
```

**Comment by SidneyAllen** [comment]
Hi @dldobbin,

I've released 2.3.13 which handles non ApiExceptions like IOExceptions and returns them as XeroApiExceptions.

I've  tested using the code above and confirmed XeroApiException xae is caught.
